src/ServiceBus.AttachmentPlugin.Tests/ApprovalFiles/ApiApprovals.AzureStorageAttachmentPlugin.approved.txt

```diff
@@ -11,6 +11,8 @@ namespace Microsoft.Azure.ServiceBus
     }
     public static class AzureStorageAttachmentConfigurationExtensions
     {
+        public const string DefaultMessagePropertyToIdentitySasUri = "$attachment.sas.uri";
+        public static readonly System.TimeSpan DefaultSasTokenValidationTime;
```

Comment:
What's the value of exposing this? If the default is documented and known, how you would use this publicly exposed member field in your code? I'd like to understand how it would be used in the customer/user code.

Reply:
The argument is the same as on [#185](https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/issues/185#issuecomment-610738095)

> Focusing on `ContainerName` and `MessagePropertyToIdentifyAttachmentBlob` let's assume you're using the defaults. Both properties, `ContainerName` and `MessagePropertyToIdentifyAttachmentBlob` have their defaults provided by the plugin which you'd need to use. Those are [well documented](https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/blob/develop/README.md#configure-blob-container-name) and will **not change** to keep wire compatibility. You can use those strings in your code. My understanding is you don't like to use the strings and rather have something strongly types exposed by the `AzureStorageAttachmentConfiguration` class. That can be done, don't see an issue with it.

I would like to have access to a strongly typed values, so I don't have to worry about the values, even though I *know* they won't change. It reduces complexity (albeit not much) in my code.

I hope it makes sense.

Reply:
> I would like to have access to a strongly typed values, so I don't have to worry about the values, even though I know they won't change.

Absolutely. That's why it makes sense to have access to the defaults of the SAS URI property, the container name, and the blob identifier property. I can trace how you'd be using that in your code if you need to access the SAS URI property/container/message property for blob name. What I don't see is how the default SAS token validation time is used?

>  It reduces complexity (albeit not much) in my code.

@sorenhansen, could you show please demonstrate how `AzureStorageAttachmentConfigurationExtensions.DefaultSasTokenValidationTime` would help with that?

Reply:
@SeanFeldman Sorry, I didn't catch that you specifically meant `DefaultSasTokenValidationTime`

I agree that making `DefaultSasTokenValidationTime` public could be a bit much. It just made sense to me to make all default values public in case there would be a use case for them. In my particular case, I only need `DefaultMessagePropertyToIdentitySasUri`.

I guess you would rather leave `DefaultSasTokenValidationTime` internal for now, and then only expose it when it's absolutely needed?

Reply:
Yeah, I've been trained over the time not to reveal anything that is not absolutely necessary. Once it's public, you need to worry about deprecation and migration. When it stays internal, can do whatever is needed with it.

I would suggest removing that bit only. It's still good to mention the default value in the documentation for readers to have an idea. Additionally, some XML documentation can be added to the code, on the parameter, mentioning the default value here:
https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/blob/7dfff04543faaa476f73a2d8e49e50b8dee61d5b/src/ServiceBus.AttachmentPlugin/AzureStorageAttachmentConfigurationExtensions.cs#L18

Reply:
@sorenhansen, will you do the update or should I?

Reply:
@SeanFeldman I'll do the update. Just been busy the last few days 🙂  

I plan on doing it later today

Reply:
@SeanFeldman I've updated the code, making `DefaultSasTokenValidationTime` internal again, and documenting the default value in the `AzureStorageAttachmentConfigurationExtensions.WithBlobSasUri` method.

Reply:
Thank you. I have reviewed and left a comment.
Sorry for nitpicking. Don't want to release your good changes without ensuring we're not causing more work to the users.

Reply:
You're not at all nitpicking 🙂
When I made `DefaultMessagePropertyToIdentitySasUri` I was a bit unhappy with it being in another class than the other constants, so I'm glad you pushed a refactor.

I've updated the PR with your proposed changes. I've left `DefaultSasTokenValidationTime` in `AzureStorageAttachmentConfigurationExtensions` as it's still `internal`